src/service/auth-service.js

```diff
@@ -0,0 +1,117 @@
+const ipLib = require('ip');
+const jwtUtil = require('./util/jwt-util');
+
+module.exports = class AuthService {
+    constructor(ctx) {
+        this._authConfig = ctx.config.auth;
+        this._repository = ctx.repositoryModuleManager;
+    }
+
+    /**
+     * Authenticate users based on provided ip and token
+     * @param ip
+     * @param token
+     * @returns {boolean}
+     */
+    async authenticate(ip, token) {
+        const isWhitelisted = this._isIpWhitelisted(ip);
+        const isTokenValid = this._isTokenValid(token);
+
+        if (!isWhitelisted || !isTokenValid) {
+            return false;
+        }
+
+        return !(await this._isTokenRevoked(token));
```

Comment:
Why did you create two methods _isTokenValid and _isTokenRevoked, i would expect that valid function should also check revoked status

Reply:
Good point, I had dilemma whether `isTokenRevoked` should be part of `isValid` function.

The argument for separating two was that `isValid` only checks validity of provided token. This means that for JWT it checks whether we are the one who generated it and that it hasn't expired yet (both information are provided by the nature of JWT. Later on, this method will be extended with validation for provided wallet.

Although token is valid, it might be revoked and that's why ended up splitting the logic of into two methods.

But again, token is not valid since it is revoked, so this might also be part of `isValid` method.

Honestly, I am still not sure about the choice made here 😄

Reply:
But since you expected that `isTokenRevoked` is a part of `isValid` I guess this separation was result of overthinking and therefore we should merge `isTokenRevoked` into `isValid`